### DISCUSSION TITLE
Color for current line and visual blocks missing

### DISCUSSION THREAD

**Discussion by janhejzl** [arcticicestudio/nord#208] [Help & Support]
Despite I have set `cursoline` highlighting up, there is no visible color. The same with visual block selections. The only clue about what is being selected is thanks to vim-indentline chars disappearing. (In gVim it works perfectly.)

- it is run in _terminal_
- bash 5.0.3
- terminal 3.32.1-1ubuntu1
- in terminal I tried to use Nord theme via Gogh or the basic system theme (_it seems this may be the problem, it somehow overrides the colors_)
- it is run alone, without any multiplexer
- Ubuntu 19.04
- as I can remember, other themes were OK (gruvbox, space_vim_dark, etc.)

At the moment, I have no idea how to run bare-bones Vim with the nord theme without broking my .vim directory.

This is what I get with both nord theme in terminal and in vim (besides the current line highlighting problem, it seems all colors are distorted a bit too):
![obrazek](https://user-images.githubusercontent.com/8298610/63495725-b3c04d00-c4c0-11e9-81a0-91d2736b5d45.png)

**Comment by arcticicestudio** [comment]
Hi @welblaud, I need some more information regarding your setup (like mentioned in the issue template), otherwise there's no way trying to reproduce the problem:

- Are you running in _GUI_ or _terminal_ mode?
  - If you're running in _terminal_ mode, what's the name and version of your shell and terminal?
  - Are you within an terminal multiplexer like `tmux` or connected to an external system via SSH?
- What is the operating system and version you're running?
- Have you tried to reproduce the problem with a clean Vim configuration (e.g. no other plugins and customizations) that only loads and sets Nord as `colorscheme`?
  - If not, have you checked that there are no any other modifications in your `vimrc` that might cause this problem?
- Is the problem reproducible using another theme?

Also, if you're using the `st` terminal the problem might be related to nordtheme/vim#166. If so please try to follow the steps that were done in the comments of nordtheme/vim#166 (setting the special variables) to see if this solves the problem.

**Comment by Benjamin-Lee** [comment]
I am currently facing these issues running iTerm with Fish on MacOS, so it's definitely not an isolated issue.

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/13618205/64211669-e2e99d80-ce74-11e9-956e-06529c116b6b.png">

**Comment by dagolinuxoid** [comment]
Terminal: simple terminal [st](https://st.suckless.org/)
Theme: [patch nordtheme](https://st.suckless.org/patches/nordtheme/)
Application: Vim version 8.2.24
Colorscheme: [nord](https://github.com/arcticicestudio/nord-vim)
OS: ElementaryOS 
Usually I don't use line highlighting but I tested it via `:set cursorline`  and it works as expected:
[![nord.png](https://i.postimg.cc/HnrYTwC9/nord.png)](https://postimg.cc/5QMd3FyX)
PS.

> Note that setting the 'cursorline' or 'cursorcolumn' options can cause Vim to respond slowly, especially for large files or files with long lines.

**Comment by arcticicestudio** [comment]
@dagolinuxoid Thanks for testing and sharing your results, looks like both problems are related to _Vim_'s color space again.

@Benjamin-Lee @welblaud Could you please try to add the following option to your configuration? It looks like Vim uses the 256 color palette instead of the hexadecimal colors defined by _Nord_.

```vim
if (has("termguicolors"))
  set termguicolors
endif
```

**Comment by dagolinuxoid** [comment]
BTW unfortunately in some cases in order to enable true color support (make sure your terminal support it),  using just `set termguicolors` isn't enough
I managed to make it work by adding these lines into vimrc config file:
```vim
" set Vim-specific sequences for RGB colors | enable true colors
let &t_8f = "\<Esc>[38;2;%lu;%lu;%lum"
let &t_8b = "\<Esc>[48;2;%lu;%lu;%lum"
set termguicolors
```

**Comment by galah92** [comment]
I've got the same problem using Windows Terminal. Setting`set termguicolors` does work and `set cursorline` now works, but now the integrated terminal in vim (using `:term`) has the wrong colors. Am I missing something?

**Comment by ManuelSaavedraSoto** [comment]
I'd add the lines to the install guide so people are aware of the issue

**Comment by BenResTech** [comment]
Confirming that adding the `set termguicolors` and `set cursorline` options makes this work for me. This really would be great to add to the installation instructions.

**Comment by jvoisin** [comment]
It's [documented in the readme]( https://github.com/arcticicestudio/nord-vim/blob/develop/README.md?plain=1#L13) that only _true colours_ are supported.

**Comment by svengreb** [comment]
Thank you for your patience! 🙏🏼
It‘s been a while [since I had free time to focus more on Nord, and my open source projects in general](https://github.com/orgs/nordtheme/discussions/183#user-content-todays-scope), and invest time in this issue due to [work-life balance](https://github.com/orgs/nordtheme/discussions/183#work-life-balance).

I recently published the first [“Northern Post — The state and roadmap of Nord“ announcement](https://github.com/orgs/nordtheme/discussions/183) which includes all details about the plans and future of the _Nord_ project, including [the goal of catching up with the backlog](https://github.com/orgs/nordtheme/discussions/183#catching-up-on-the-backlog). This issue is part of the backlog and therefore I want to triage and process it to get one step closer to a “clean state“. Read the [announcement about reaching the “clean“ contribution triage state](https://github.com/orgs/nordtheme/discussions/202) in Nord‘s discussions for more details about the goal.

Due to the scope of this issue of not being “purely technical“, or being related to the project‘s code at all, I‘ll convert it into a discussion for [Nord‘s central community hub on GitHub](https://github.com/orgs/nordtheme/discussions). This way the repository is easier to maintain by separating conversations about the actual code while also making help requests, ideas and other contributions from users more visible and supportable by the community.

@welblaud @Benjamin-Lee Since there has been some time without further communication, it would be great if you could quickly give some feedback if the problem has been “resolved“ in the meantime using [the mentioned `termguicolors` option](https://github.com/orgs/nordtheme/discussions/208#discussioncomment-6032232).

**Reply by jvoisin** [reply]
@svengreb happy to help/take ownership of the vim theme if you want, I'd love to [put my changes upstream](https://github.com/jvoisin/nord-vim)
